### ISSUE TITLE
How to remove sensitive information (ex: passwords) from the output

### ISSUE THREAD

**Issue by andrecarlucci** [fluentmigrator/fluentmigrator#1242]
Hi guys,

I have one migration to create a user on Postgres:

```
var pass = Environment.GetEnvironmentVariable("GENERATED_PASSWORD");
Execute.Sql($"CREATE USER bob WITH PASSWORD '{pass}';");
```
The output of this migration would be:
```
-------------------------------------------------------------------------------
1: _001_Change_MasterUser migrating
-------------------------------------------------------------------------------
Beginning Transaction
ExecuteSqlStatement CREATE USER bob WITH PASSWORD 'jw8s0F4dkjskfjkjsdfahu38**3e;;d';
Committing Transaction
1: _001_Change_MasterUser migrated
```
We use this for our pipelines, we let the developers create databases on-demand in lower environments but we take control of the password for the production ones.

For that reason, we don't want the new password to be exposed in the log files.

Is there a way to achieve this?

**Expected benefit**
Sensitive information is not exposed in log files.

Tks!

**Comment by pergardebrink** [comment]
I have a related question on this. I have some migrations I generate from our legacy sql files and run them with Execute.Sql() and the output becomes cluttered with all SQL statements.

As I can see, the issue is that ExecuteSqlStatementExpression overrides ToString() that concatenates the statement, which is then logged. https://github.com/fluentmigrator/fluentmigrator/blob/e82aafa20e6dbe3cefa221303fe23cf8bf59fffd/src/FluentMigrator.Abstractions/Expressions/ExecuteSqlStatementExpression.cs#L45
I would prefer if it would be possible to display an optional custom more descriptive message. Something like this?

```
public override void Up() 
{
    Exeucte.Sql("UPDATE Something SET x=1", "Update Something");
}
```

**Comment by jzabroski** [comment]
@andrecarlucci First, Great to see you decided to join FluentMigrator as a community member.  I'm glad I reached out.  Sorry for the delays in responding - COVID-19 lockdown really slows down life with things like buying grocercies, etc.  

Second, I think you bring a good point.  I have not had any GDPR requirements but this is definitely huge for GDPR customers.  And, on my project the last 3 years, we use SQL Server so we just manage users as Windows AD Groups, so I have not hit password creation issues.

@pergardebrink So, here's my thoughts on adding a Description field to override outputting SQL:

1.  Technically it shouldn't break anyone API wise unless they're using some weird reflection and expecting only a single method called Sql.
2. We used to have a framework for "announcing" what was getting executed, but we got rid of it when Microsoft introduced Microsoft.Extensions.Logging - it was just a lot of extra code to maintain (Technically, IAnnouncer, TextWriterAnnouncer, etc. all still exist, but they're marked obsolete in v3).
3. As part of removing the Announcer abstraction, we did lose a degree of freedom in consolidating and formatting log messages.  For example, if you chain Insert.Into statements, it sort of misreports the number of inserts (it reports the number of FluentMigrator Insert.Into root expressions, as opposed to records inserted).  I say "sort of misreports", because it's rather nice to just log you're inserting data without writing a row for every insert.  That said, I bring this up because today the user doesn't have an easy way to choose what level of detail to print.
4. Similarly, as part of deprecating Announcer, Mark cleaned up a lot of the output noise and kept it as simple as possible.  Coming from 1.6.2 to 3.2.1, this was a bit of a shock at first but I think stylistically its correct choice.
5. (Most importantly) *As long as we regression test that this change doesn't break `--no-connection` and the ShowSql setting people rely on to print SQL files for DBAs to review, I don't see this as a bad change.*  Your test is just a unit test - it doesn't cover any regression test.  I think it should pass, but it would be bad if we merged this and broke a key feature like that.
6. I'm currently getting closer and closer to shipping 3.3.0, at which point I can realistically merge this.

@andrecarlucci What do you think?

**Comment by pergardebrink** [comment]
@jzabroski, first of all wanted to express my thanks for all your efforts in this project! I understand that times as these are tough, and taking time to work on open source projects like this on your spare time should come second after managing life in these COVID-times! 

I have just started testing fluent migrator and it seems to solve a lot of the issues I've had with the current migration tooling, and can replace a bunch of old hard-to-read custom bash scripts! 

I updated the pull requests with the test you talked about, so please let me know if there's more I should add or change. I will hopefully be contributing more to this project in the future if there's things I can do!

**Comment by jzabroski** [comment]
@pergardebrink Yeah, FluentMigrator is awesome and to think, when my current job insisted I use it, I was skeptical.  Now I love it.  It has bailed me out of some MAJOR jams that were impossible to solve before or would delay a project a full week to solve.

For example, converting all ntext fields in SQL Server db to nvarchar(max) is trickier than you would think due to needing to write an update command for every row in the table in order to "force" the db engine to "touch" the DB page and move the BLOB to a spot that actually is "seen" by the engine for things like wild card queries.

As far as your PR goes, I am off next week and while I plan to take it easy with minimal computer time, I look forward to merging this.  For one, I asked @andrecarlucci to abandon his [SharpMigrations](https://github.com/sharpmigrations/sharpmigrations) project and come help here around a year ago.  For two, this PR really addresses half of an open problem I had, which was described above.  It's not a perfect PR but I think it's sufficient enough, and minimal enough, that it shouldn't cause headaches down the road (bite my tongue).

**Comment by andrecarlucci** [comment]
Hi guys! Sorry about the delay. I guess we are all busy these days :)
@pergardebrink PR should solve my problem for now. Thanks a lot!

**Comment by jzabroski** [comment]
@pergardebrink  Solution is merged.  Should release 3.2.7 this week.

**Comment by jzabroski** [comment]
@andrecarlucci @pergardebrink  Fixed in 3.2.7 released this morning. Thank you for your help and bringing this to my attention and coming up with a simple, elegant solution.
